ipalib/x509.py

```diff
@@ -145,6 +148,23 @@ def load_certificate_list_from_file(filename):
         return load_certificate_list(f.read())
 
 
+def pkcs7_to_pems(data, datatype=PEM):
+    """
+    Extract certificates from a PKCS #7 object.
+
+    Return a ``list`` of X.509 PEM strings.
+
+    May throw ``ipautil.CalledProcessError`` on invalid data.
+
+    """
+    cmd = [
+        paths.OPENSSL, "pkcs7", "-print_certs",
```

Comment:
The import was removed for a reason, putting it back will break the `ipalib` package on PyPI.

Now that we require `pyasn1_modules`, we can use PyASN1 to do the parsing instead:
```python
    if datatype == PEM:
        match = re.search(r'-----BEGIN PKCS7-----(.*?)-----END PKCS7-----', data, re.DOTALL)
        if not match:
            raise ValueError()
        data = base64.b64decode(match.group(1))

    ci, _rest = decoder.decode(data, rfc2315.ContentInfo())
    if ci['contentType'] != rfc2315.ContentType('1.2.840.113549.1.7.2'):
        raise ValueError()

    data = str(ci['content'])
    sd, _rest = decoder.decode(data, rfc2315.SignedData())

    certs = [encoder.encode(c['certificate']) for c in sd['certificates']]
    return certs
```